### ISSUE TITLE
Main Application would need more configuration options

### ISSUE THREAD

**Issue by nagkumar** [GaetanoPiazzolla/springboot-testcontainers#3]
Currently

`gradle bootRun`

fails to start

```
C:\temp\springboot-testcontainers>gradle bootRun

> Task :bootRun

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::       (v3.1.1-SNAPSHOT)

2023-06-13T09:31:43.380+05:30  INFO 4756 --- [           main] gae.piaz.springtc.Application            : Starting Application using Java 19.0.2 with PID 4756 (C:\temp\springboot-testcontainers\build\classes\java\main started by nagku in C:\temp\springboot-testcontainers)
2023-06-13T09:31:43.382+05:30  INFO 4756 --- [           main] gae.piaz.springtc.Application            : No active profile set, falling back to 1 default profile: "default"
2023-06-13T09:31:43.833+05:30  INFO 4756 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Multiple Spring Data modules found, entering strict repository configuration mode
2023-06-13T09:31:43.835+05:30  INFO 4756 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data JPA repositories in DEFAULT mode.
2023-06-13T09:31:43.940+05:30  INFO 4756 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 98 ms. Found 1 JPA repository interfaces.
2023-06-13T09:31:43.952+05:30  INFO 4756 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Multiple Spring Data modules found, entering strict repository configuration mode
2023-06-13T09:31:43.953+05:30  INFO 4756 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data Redis repositories in DEFAULT mode.
2023-06-13T09:31:43.960+05:30  INFO 4756 --- [           main] .RepositoryConfigurationExtensionSupport : Spring Data Redis - Could not safely identify store assignment for repository candidate interface gae.piaz.springtc.data.CustomerRepository; If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository
2023-06-13T09:31:43.960+05:30  INFO 4756 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 2 ms. Found 0 Redis repository interfaces.
2023-06-13T09:31:44.404+05:30  INFO 4756 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8181 (http)
2023-06-13T09:31:44.412+05:30  INFO 4756 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2023-06-13T09:31:44.413+05:30  INFO 4756 --- [           main] o.apache.catalina.core.StandardEngine    : Starting Servlet engine: [Apache Tomcat/10.1.9]
2023-06-13T09:31:44.491+05:30  INFO 4756 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2023-06-13T09:31:44.492+05:30  INFO 4756 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 1062 ms
2023-06-13T09:31:44.677+05:30  WARN 4756 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'dataSourceScriptDatabaseInitializer' defined in class path resource [org/springframework/boot/autoconfigure/sql/init/DataSourceInitializationConfiguration.class]: Unsatisfied dependency expressed through method 'dataSourceScriptDatabaseInitializer' parameter 0: Error creating bean with name 'dataSource' defined in class path resource [org/springframework/boot/autoconfigure/jdbc/DataSourceConfiguration$Hikari.class]: Failed to instantiate [com.zaxxer.hikari.HikariDataSource]: Factory method 'dataSource' threw exception with message: Failed to determine a suitable driver class
2023-06-13T09:31:44.686+05:30  INFO 4756 --- [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2023-06-13T09:31:44.697+05:30  INFO 4756 --- [           main] .s.b.a.l.ConditionEvaluationReportLogger :

Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2023-06-13T09:31:44.709+05:30 ERROR 4756 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   :

***************************
APPLICATION FAILED TO START
***************************

Description:

Failed to configure a DataSource: 'url' attribute is not specified and no embedded datasource could be configured.

Reason: Failed to determine a suitable driver class


Action:

Consider the following:
        If you want an embedded database (H2, HSQL or Derby), please put it on the classpath.
        If you have database settings to be loaded from a particular profile you may need to activate it (no profiles are currently active).


> Task :bootRun FAILED

FAILURE: Build failed with an exception.


```
it would be nice to include the Redis, kafka server URL properties as part of main application.properties.

By seeing the production way of running spring boot vs testcontainers version, any can get the value and simplicity of testcontainers.

**Comment by GaetanoPiazzolla** [comment]
running the app normally would require a redis + kafka + postgres + python flask instance running somewhere and we need to configure all the URLs. it's out of the scope of the article and it's useless.

**Comment by nagkumar** [comment]
At least highlight what are the main code parts and API, that shall utilize testcontainers.. unless the real code exists <at the URLs to connect> it is difficult for any to understand what API in main are now using testcontainers servers...

I tried to look at the main, not very clear, which API are related to kafka, redis as I hardly see any related calls. As it is integration, some close to real code can really help..pl. rethink. else it looks like test code is everything.. 

Basically, I wish this example clearly says what URLs change..
